web/angular-wallet/src/app/setup/account/account.service.ts

```diff
@@ -245,12 +245,10 @@ export class AccountService {
   */
   public synchronizeAccount(account: Account): Promise<Account> {
     return new Promise(async (resolve, reject) => {
-      await Promise.all([
-          this.syncAccountDetails(account),
-          this.syncAccountTransactions(account),
-          this.syncAccountUnconfirmedTransactions(account),
-        ]
-      );
+      await this.syncAccountDetails(account);
+      await this.syncAccountTransactions(account);
+      await this.syncAccountUnconfirmedTransactions(account);
```

Comment:
These methods modify the same object under the hood, so this is a race condition.